task8/src/main/java/PostServlets/One.java

```diff
@@ -0,0 +1,65 @@
+package PostServlets;
+import PostWork.Service;
+import PostWork.Post;
+import com.google.gson.Gson;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+
+public class OnePost extends HttpServlet {
+
+    private Service posts = Service.getInstance();
+
+    @Override
+    protected void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String str = request.getQueryString();
+        String sID = str.substring(3, str.length());
+        int id = Integer.parseInt(sID);
+        response.getOutputStream().println("it Works id= " + id);
+        if (posts.removePost(id)) {
+            response.getOutputStream().println("deleted\n");
+        } else {
+            response.getOutputStream().println("Not found\n");
+        }
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String str = request.getQueryString();
+        String sID = str.substring(3, str.length());
+        int id = Integer.parseInt(sID);
+        Post post = posts.getPost(id);
+
+        if (post != null) {
+            response.getWriter().print((new Gson()).toJson(posts.getPost(id)));
+        } else {
+            response.getOutputStream().println("Not found\n");
```

Comment:
That's OK, but consider changing the status code as well. To be in line with the REST format